app/services/data_migrations/remove_duplicate_provider.rb

```diff
@@ -0,0 +1,43 @@
+module DataMigrations
+  class RemoveDuplicateProvider
+    TIMESTAMP = 20210828151830
+    MANUAL_RUN = false
+
+    def change
+      permissions = ProviderRelationshipPermissions
+        .joins('INNER JOIN providers tp ON training_provider_id = tp.id')
+        .joins('INNER JOIN providers rp ON ratifying_provider_id = rp.id')
+        .where('tp.name = rp.name')
+
+      if permissions.any?
+        permissions.each do |permission|
+          # Do nothing if the ratifying provider runs courses
+          next if permission.ratifying_provider.courses.any?
+          # Do nothing if the ratifying provider has additional org relationships
+          next if ProviderRelationshipPermissions
+            .where(ratifying_provider: permission.ratifying_provider)
+            .where.not(training_provider: permission.training_provider).any?
+
+          courses = permission.training_provider.courses.where(accredited_provider: permission.ratifying_provider)
+
+          # Do nothing if the ratifying provider ratifies other courses
+          next if (permission.ratifying_provider.accredited_courses - courses).any?
+
+          ActiveRecord::Base.transaction do
+            courses.each do |course|
+              course.update!(accredited_provider: nil)
+            end
+
+            permission.destroy!
+
+            # Ensure we don't orphan any users who were only associated with the duplicate provider
+            providers_for_ratifying_provider_users = permission.ratifying_provider.provider_users.map(&:providers).flatten
```

Comment:
I wonder if we should check per user rather than all providers for all users, to avoid orphaning some but not others? (I did a dry run and doesn't look like we will though)

Reply:
Whats the plan after we run this migration, are we planning on contacting the providers/users?